public/app/plugins/panel/datagrid/DataGridPanel.tsx

```diff
@@ -115,7 +133,10 @@ export function DataGridPanel({ options, data, id, fieldConfig, width, height }:
       values.push(null);
       return { ...f, values };
     });
-    publishSnapshot({ ...frame, fields, length: frame.length + 1 }, id);
+
+    if (onUpdateData) {
+      onUpdateData([{ ...frame, fields, length: frame.length + 1 }]);
+    }
```

Comment:
maybe have a util function name updateDataGridData(frame, panelContext.onUpdateData) , that would handle checking if onUpdateData exists and turning single frame into an array. So many places that check if(onUpdateData) ...

Reply:
I was wodnering the same about all these if conditions